src/editor/reducers/notebook-reducer.js

```diff
@@ -161,16 +161,12 @@ const notebookReducer = (state = newNotebook(), action) => {
         lastSaved: new Date().toISOString(),
         notebookInfo: Object.assign({}, state.notebookInfo, {
           user_can_save: true,
-          revision_id: action.newRevisionId
+          revision_id: action.newRevisionId,
+          revision_is_latest: true
         })
       });
     }
 
-    case "UPDATE_CHECKING_NOTEBOOK_REVISION_IS_LATEST": {
```

Comment:
With the server-side checking of revision integrity on submit, we no longer the complexity of maintaining this state on the client. This was a pretty narrow and unlikely edge case in any event.